libs/langgraph/langgraph/pregel/loop.py

```diff
@@ -294,29 +296,17 @@ def put_writes(self, task_id: str, writes: Sequence[tuple[str, Any]]) -> None:
         """Put writes for a task, to be read by the next tick."""
         if not writes:
             return
+        checkpoint_during = self.checkpoint_during or any(w[0] == TASKS for w in writes)
         # deduplicate writes to special channels, last write wins
         if all(w[0] in WRITES_IDX_MAP for w in writes):
             writes = list({w[0]: w for w in writes}.values())
+        # remove existing writes for this task
+        self.checkpoint_pending_writes = [
```

Comment:
Dumb q but this is diffrent than before in the case where there's multiple writes for a single WRITES_IDX_MAP channel - did we add an invarian where we know it won't be duplicated here?

Reply:
I think that's covered by line 302 above